Source/WebKit/UIProcess/API/haiku/WebView.cpp

```diff
@@ -62,7 +62,14 @@ BWebView::BWebView(BRect frame, BWindow* myWindow)
     WKPageConfigurationSetContext(config.get(), fContext.get());
 
     fViewPort=adoptWK(WKViewCreate("Webkit", frame, myWindow, config.get()));
-    fAppLooper = myWindow->Looper(); 
+}
+
+void BWebView::initialize()
+{
+    if (BLooper::LooperForThread(find_thread(NULL)) == NULL)
+        debugger("BWebView::initialize must be called from a running BApplication (e.g. this cannot be called from BApplication's constructor).");
```

Comment:
is this not fixed after your changes to RunLoop to detect this and attach the looper to the main thread anyways, even if the BApplication is not running?